src/components/input/SelectNext.tsx

```diff
@@ -213,7 +267,7 @@ function SelectMain<T>({
   useArrowKeyNavigation(listboxRef, {
     horizontal: false,
     loop: false,
-    autofocus: true,
+    autofocus: !nativePopoverSupported,
```

Comment:
Automatically focusing the corresponding option results in the item being focused before we had the chance to reposition the popover, causing the page to scroll to the very top.

Instead, when the browser supports popovers natively, we do not focus the item as part of `useArrowKeyNavigation`, but via `toggle` event on the listbox. See https://github.com/hypothesis/frontend-shared/pull/1540/files#diff-92242547c6b4471729edbe9319b52f8bf068f28ed9a0d6ed681a4cdeba48e777R153